### PR TITLE
add osin repository to CI

### PR DESCRIPTION
**Pull request by stlaz**
/cc @marun @vareti

### CODE REVIEW

File: ci-operator/config/openshift/osin/openshift-osin-master.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    namespace: openshift
+    tag: "7"
+binary_build_commands: go build ./...
+build_root:
+  image_stream_tag:
+    cluster: https://api.ci.openshift.org
+    name: release
+    namespace: openshift
+    tag: golang-1.13
+promotion:
```
Comment (tnozicka) [inline ci-operator/config/openshift/osin/openshift-osin-master.yaml:14]: what image is this promoting? Aren't you missing config for image build here?

File: ci-operator/config/openshift/osin/openshift-osin-master.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+base_images:
+  os:
+    cluster: https://api.ci.openshift.org
+    name: centos
```
Comment (tnozicka) [inline ci-operator/config/openshift/osin/openshift-osin-master.yaml:4]: if possible, I'd base this on the common base image

File: ci-operator/config/openshift/osin/openshift-osin-master.yaml (hunk 1 above)
Comment (stlaz) [reply ci-operator/config/openshift/osin/openshift-osin-master.yaml:14]: I blatantly copied this from `apiserver-library-go` as it's also a library. If this section is not meant to be here, shall I remove it from there as well (in a different PR, perhaps)?

File: ci-operator/config/openshift/osin/openshift-osin-master.yaml (hunk 2 above)
Comment (stlaz) [reply ci-operator/config/openshift/osin/openshift-osin-master.yaml:4]: this is the common image for all of apiserver-library-go and library-go, should a completely different config be used here?

File: ci-operator/config/openshift/osin/openshift-osin-master.yaml (hunk 1 above)
Comment (tnozicka) [reply ci-operator/config/openshift/osin/openshift-osin-master.yaml:14]: I suppose the automation brought it back. It doesn't hurt to leave it there.

Comment (petr-muller) [reply ci-operator/config/openshift/osin/openshift-osin-master.yaml:14]: We use "promotion to ocp:4.X" to recognize official repos for the tooling that do branching, fast-forwards, private org mirrors etc, so it is desirable to have this if this should be involved in all these processes.

**Pull request merged**